src/Interpreters/SystemLogStorage.cpp

```diff
@@ -0,0 +1,341 @@
+#include <Interpreters/SystemLogStorage.h>
+
+#include <Interpreters/AsynchronousMetricLog.h>
+#include <Interpreters/CrashLog.h>
+#include <Interpreters/MetricLog.h>
+#include <Interpreters/OpenTelemetrySpanLog.h>
+#include <Interpreters/PartLog.h>
+#include <Interpreters/QueryLog.h>
+#include <Interpreters/QueryThreadLog.h>
+#include <Interpreters/QueryViewsLog.h>
+#include <Interpreters/SessionLog.h>
+#include <Interpreters/TextLog.h>
+#include <Interpreters/TraceLog.h>
+#include <Interpreters/FilesystemCacheLog.h>
+#include <Interpreters/FilesystemReadPrefetchesLog.h>
+#include <Interpreters/ProcessorsProfileLog.h>
+#include <Interpreters/ZooKeeperLog.h>
+#include <Interpreters/TransactionsInfoLog.h>
+#include <Interpreters/AsynchronousInsertLog.h>
+#include <Interpreters/BackupLog.h>
+
+#include <Interpreters/Context.h>
+#include <Interpreters/InterpreterCreateQuery.h>
+#include <Interpreters/InterpreterInsertQuery.h>
+#include <Interpreters/InterpreterRenameQuery.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTIndexDeclaration.h>
+#include <Parsers/ASTInsertQuery.h>
+#include <Parsers/ASTRenameQuery.h>
+#include <Parsers/ParserCreateQuery.h>
+#include <Parsers/formatAST.h>
+#include <Parsers/parseQuery.h>
+#include <Processors/Executors/PushingPipelineExecutor.h>
+#include <Common/SystemLogBase.h>
+#include <Common/ActionLock.h>
+#include <Common/logger_useful.h>
+#include <Storages/IStorage.h>
+#include <Storages/MergeTree/MergeTreeSettings.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int NOT_IMPLEMENTED;
+}
+
+namespace ActionLocks
+{
+    extern const StorageActionBlockType PartsMerge;
+}
+
+namespace
+{
+
+class StorageWithComment : public IAST
```

Comment:
What's the difference with `src/Interpreters/SystemLog.cpp` ?

Reply:
>but I consider this refactoring to be a subject of a separate PR

I'm not the one who decides, but I think it will be easier for the reviewer if there will be a tiny change in the existing class instead of copy-paste.